src/extent.c

```diff
@@ -468,10 +467,9 @@ extents_first_fit_locked(tsdn_t *tsdn, arena_t *arena, extents_t *extents,
 		 */
 		if (extents->delay_coalesce &&
 		    (sz_pind2sz(i) >> opt_lg_extent_max_active_fit) > size) {
-			size_ok = false;
+			break;
 		}
-		if (size_ok &&
-		    (ret == NULL || extent_snad_comp(extent, ret) < 0)) {
+		if (ret == NULL || extent_snad_comp(extent, ret) < 0) {
 			ret = extent;
 		}
 		if (i == SC_NPSIZES) {
```

Comment:
nit: lines 475-478 are not needed?

Reply:
Oh that’s still need, as ‘bitmap_ffu’ doesn’t accept i+1 in that case.